app/jobs/ingest_job.rb

```diff
@@ -0,0 +1,17 @@
+class IngestJob < Hyrax::ApplicationJob
+  queue_as Hyrax.config.ingest_queue_name
+
+  after_perform do |job|
+    opts = job.arguments.reverse.detect { |x| x.is_a? Hash } || {}
+    wrapper = job.arguments.first
+    ContentNewVersionEventJob.perform_later(wrapper.file_set, wrapper.user) if opts[:notification]
```

Comment:
I'm not sure what a better approach would look like, but I'm wondering if this line will get us into trouble later because of where and when this coupling is expressed. For instance, a downstream app may want to add notifications with other jobs, but we don't want to assume all jobs with `notification: true` correspond to new version events. The callbacks used before -- though invoked at the incorrect time 'til this PR -- had the advantage of giving us more flexibility, which could be overridden by downstream consumers. Can you suggest an improvement here? FWIW, I wouldn't block the PR over this, but I would like to ticket this for follow-on work.